integrations/mastodon-fetcher.md

```diff
@@ -12,17 +12,70 @@ pypi: https://pypi.org/project/mastodon-fetcher-haystack/
 repo: https://github.com/tuanacelik/mastodon-fetcher-haystack
 type: Custom Node
 report_issue: https://github.com/tuanacelik/mastodon-fetcher-haystack/issues
+logo: /logos/mastodon.png
+version: Haystack 2.0
+toc: true
 ---
-
 The `MastodonFetcher` is a simple custom component that fetches the `last_k_posts` of a given Mastodon username.
+You can see an demo of this custom component in use in the [🦄 Should I Follow?](https://huggingface.co/spaces/deepset/should-i-follow) space on Hugging Face 🤗.
+
+### **Table of Contents**
+
+- [Haystack 2.0](#haystack-20)
+  - [Installation](#installation)
+  - [Usage](#usage)
+- [Haystack 1.x](#haystack-1x)
+  - [Installation (1.x)](#installation-1x)
+  - [Usage (1.x)](#usage-1x)
+
+## Haystack 2.0
+This component expects `username` to be a complete Mastodon username. For example "tuana@sigmoid.social". If the provided username is correct and public, `MastodonFetcher` will return a list of `Document` objects where the contents are the users latest posts.
+
+### Installation  (2.0)
+```bash
+pip install mastodon-fetcher-haystack
+```
+
+### Usage (2.0)
+You can use this component on its own, or in a pipeline.
+
+#### On its own:
+```python
+from mastodon_fetcher_haystack.mastodon_fetcher import MastodonFetcher
+
+mastodon_fetcher = MastodonFetcher()
+mastodon_fetcher.run(username="tuana@sigmoid.social")
+```
+#### In a pipeline
+
+```python
+from haystack import Pipeline
+from mastodon_fetcher_haystack.mastodon_fetcher import MastodonFetcher
+from haystack.components.generators import OpenAIGenerator
+from haystack.components.builders import PromptBuilder
+
+prompt_builder = PromptBuilder(template='YOUR_PROMPT_TEMPLATE')
+llm = OpenAIGenerator(api_key'YOUR_OPENAI_API_KEY')
+
+pipe = Pipeline()
+pipe.add_component("fetcher", mastodon_fetcher)
+pipe.add_component("prompt_builder", prompt_builder)
+pipe.add_component("llm", llm)
+
+pipe.connect("fetcher.documents", "prompt_builder.documents")
+pipe.connect("prompt_builder.prompt", "llm.prompt")
+pipe.run(data={"fetcher": {"username": "tuana@sigmoid.social"}})
+```
+
+## Haystack 1.x  
 
 This component expects `query` to be a complete Mastodon username. For example "tuana@sigmoid.social". If the provided username is correct and public, `MastodonFetcher` will return a list of `Document` objects where the contents are the users latest posts.
 
-## Installation
+### Installation  (1.x)
 
-Run `pip install mastodon-fetcher-haystack` to install the latest available release.
+Run `pip install mastodon-fetcher-haystack==0.0.1`
```

Comment:
Can you turn this one into a code block? This way a small `Copy` button appears 👇 
```bash
pip install mastodon-fetcher-haystack==0.0.1
```

Reply:
Also, feel free to add more explanation here as we're using the same package for both versions of Haystack:
```
The latest versions of `mastodon-fetcher-haystack` are compatible only with Haystack 2.x. You need to specify the version explicitly to import the `MastodonFetcher` component suitable with Haystack 1.x.
```